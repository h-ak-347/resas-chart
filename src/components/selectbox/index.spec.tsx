import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import SelectBox from './index';

const mockDefaultProps: React.ComponentProps<typeof SelectBox> = {
  selectedPrefectures: [{ prefCode: 1, prefName: '北海道' }],
  prefectures: [
    {
      prefCode: 1,
      prefName: '北海道',
    },
    {
      prefCode: 2,
      prefName: '青森県',
    },
  ],
  clickHandler: jest.fn(),
};

describe('SelectBox', () => {
  it('should match to snapshot.', async () => {
    const { asFragment } = render(<SelectBox {...mockDefaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display modal window when click the select-box.', async () => {
    render(<SelectBox {...mockDefaultProps} />);
    expect(screen.getByRole('dialog')).toHaveAttribute('data-is-show', 'false');
    await userEvent.click(screen.getByRole('button', { name: '北海道' }));
    expect(screen.getByRole('dialog')).toHaveAttribute('data-is-show', 'true');
  });
});
