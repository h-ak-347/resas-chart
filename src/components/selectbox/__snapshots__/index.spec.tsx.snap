// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox should match to snapshot. 1`] = `
<DocumentFragment>
  <div>
    <button
      class="selectBox__toggleButton"
      type="button"
    >
      <span
        class="selectBox__selected"
      >
        北海道
      </span>
    </button>
    <div
      class="selectBox__modal"
      data-is-show="false"
      role="dialog"
    >
      <div
        class="selectBox__modalInner"
      >
        <ul
          class="selectBox__list"
        >
          <li>
            <label
              for="prefCheckbox1"
            >
              <input
                checked=""
                id="prefCheckbox1"
                name="北海道"
                type="checkbox"
                value="1"
              />
              <span
                class="selectBox__labelText"
              >
                北海道
              </span>
            </label>
          </li>
          <li>
            <label
              for="prefCheckbox2"
            >
              <input
                id="prefCheckbox2"
                name="青森県"
                type="checkbox"
                value="2"
              />
              <span
                class="selectBox__labelText"
              >
                青森県
              </span>
            </label>
          </li>
        </ul>
      </div>
      <button
        aria-label="close"
        class="selectBox__closeButton"
      />
    </div>
  </div>
</DocumentFragment>
`;
